hazelcast/src/main/java/com/hazelcast/sql/impl/QueryId.java

```diff
@@ -35,20 +35,20 @@
     private long memberIdLow;
 
     /** Local ID: most significant bits. */
-    private long localHigh;
+    private long localIdHigh;
 
     /** Local ID: least significant bits. */
-    private long localLow;
+    private long localIdLow;
 
     public QueryId() {
         // No-op.
     }
 
-    QueryId(long memberIdHigh, long memberIdLow, long localHigh, long localLow) {
+    public QueryId(long memberIdHigh, long memberIdLow, long localIdHigh, long localIdLow) {
         this.memberIdHigh = memberIdHigh;
         this.memberIdLow = memberIdLow;
-        this.localHigh = localHigh;
-        this.localLow = localLow;
+        this.localIdHigh = localIdHigh;
```

Comment:
Both serialization service and client protocol have built-in UUID support. If the only purpose of these High and Low part splits is to support them, I think you can just use UUID instead.

Reply:
protocol schema will redirect you if you configure it on the protocol side.
https://github.com/hazelcast/hazelcast-client-protocol/blob/master/schema/protocol-schema.json#L26

Reply:
`UUID` content is inlined into the `QueryId` to reduce the amount of litter. Real UUIDs are rarely extracted from this object. We only do `equals/hashCode` on hot paths.

Reply:
@devozerov I understand the optimization but UUID is a common class that we use. This will be an exception to how we represent it. I would prefer using UUID for the sake of readability.

Reply:
Let me highlight this once again - `QueryId` is not `UUID`. We use two `UUID`-s to guarantee its uniqueness, but other than that it is mostly an opaque blob.

Reply:
This way of this UUID usage does not match to protocol style. Not all languages use to high low split of a UUID as java.
MemberID is already created somewhere on the member, you don't need to split it for litter optimization. let the protocol encode/Decode it.
For the `localId`, how many local id's are created in the system? if they are per query, I don't think you should worry about the litter it creates. 
We already use UUID in memberId and listenerRegistrationId etc, which are also infrequent.
I understand you want to optimize it in the best way possible but protocol definition is also an important part of our architecture. Moreover, we cannot change the protocol when we publish it.

Reply:
Colleagues, I would like to emphasize one more time what is the nature of query ID. From the client perspective, it is an opaque blob that is passed back and forth. From the server perspective, this is an object that serves multiple purposes, such as 1) guaranteeing cluster-wide uniqueness; 2) providing an ability to find the originator member ID without remote requests. 

The design of the client protocol cannot enforce the content of this object, because client doesn't use any features of it, while the server does. Therefore, it cannot be `UUID`.

On the other hand, I share your concerns about future protocol changes. So I propose the other solution - to change it to `String`. This would be a good choice from the protocol perspective for an opaque object. If we would like to change the content of this object on the server, the client protocol will not be affected. 

I merged these changes to the PR, please take a look.

Reply:
@devozerov if the client doesn't need the content of this, I think 4 long value representation is much better than string. String needs to be parsed on server side and it's not a good idea. 
Our initial concern was about parsing the internal UUIDs but as you say the client won't need it.
The original QueryId should be used instead of a string.

Reply:
I used to think that QueryId was good solution, but now I think that String is better.

As explained before, the query ID is completely opaque to the client, I.e. it is only passed back and forth. String representation has two advantages over QueryId:
1. It is more stable - we can change the server-side representation without breaking protocol compatibility
1. It makes client implementation simpler. Consider that we will have SQL implemented in many clients in future. All of them would have to implement a custom object that does zero useful work

That said, IMO String is good enough. Also please note that there is no parsing in the sever side. We convert QueryId to String once (see QueryId.unparse) and them use this String as a key. There should be no negative performance impact from this. 

WDYT?